### PR TITLE
[pt2] add meta functions for `cummax` and `cummin`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #98552

### CODE REVIEW

**Comment by nkaretnikov** [comment]
There are still some skips left in `meta_dispatch_skips` due to strides being in different order.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -88,6 +88,26 @@ def meta_take(self, index, *, out=None):
...
 
 
+@register_meta([aten.cummax.default, aten.cummax.out])
+@out_wrapper("values", "indices")
+def cummax(self, dim):
+    values, indices = torch.empty_like(self), torch.empty_like(self, dtype=torch.int64)
+    if self.numel() != 0 and self.ndim != 0:
+        # Checks that dim is within bounds
+        maybe_wrap_dim(dim, self.ndim)
+    return values, indices
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:98]: numel/ndim checks are based on `cummax_out`

File: aten/src/ATen/native/ReduceOps.cpp [hunk 2]
```diff
@@ -853,10 +853,10 @@ std::tuple<Tensor, Tensor> cummin(const Tensor& self, int64_t dim) {
 }
 
 Tensor cummaxmin_backward(const Tensor& grad, const Tensor& input, const Tensor& indices, int64_t dim) {
-  if (input.numel() == 0) {
+  if (input.sym_numel() == 0) {
     return input;
   }
-  auto result = at::zeros(input.sizes(), input.options());
```
Comment (nkaretnikov) [inline aten/src/ATen/native/ReduceOps.cpp:859]: Required to unlock `test/functorch/test_aotdispatch.py`

File: torch/_meta_registrations.py (hunk 1 above)
Comment (Chillee) [reply torch/_meta_registrations.py:98]: You can maybe just use the same function for both cummax and cummin?

**Review by Chillee — APPROVED** [review]

**Pull request closed without merging**